Source/Events.Processing/Filters/RemoteFilterValidator.cs

```diff
@@ -42,10 +42,25 @@ public RemoteFilterValidator(IFetchEventsFromStreams eventsFromStreams, IFetchEv
         /// <inheritdoc/>
         public async Task Validate(IFilterProcessor<RemoteFilterDefinition> filter, CancellationToken cancellationToken)
         {
-            var streamProcessorState = await _streamProcessorStateRepository.GetOrAddNew(new StreamProcessorId(filter.Scope, filter.Definition.TargetStream.Value, filter.Definition.SourceStream), cancellationToken).ConfigureAwait(false);
+            var streamProcessorState = await _streamProcessorStateRepository.GetOrAddNew(
+                new StreamProcessorId(filter.Scope, filter.Definition.TargetStream.Value, filter.Definition.SourceStream),
+                cancellationToken)
+                .ConfigureAwait(false);
             var lastUnProcessedEventPosition = streamProcessorState.Position;
-            var artifactsFromTargetStream = await _eventTypesFromStreams.FetchTypesInRange(filter.Scope, filter.Definition.TargetStream, new StreamPositionRange(StreamPosition.Start, uint.MaxValue), cancellationToken).ConfigureAwait(false);
-            var sourceStreamEvents = lastUnProcessedEventPosition == 0 ? Enumerable.Empty<StreamEvent>() : await _eventsFromStreams.FetchRange(filter.Scope, filter.Definition.SourceStream, new StreamPositionRange(StreamPosition.Start, lastUnProcessedEventPosition - 1), cancellationToken).ConfigureAwait(false);
+            var artifactsFromTargetStream = await _eventTypesFromStreams.FetchTypesInRange(
+                filter.Scope,
+                filter.Definition.TargetStream,
+                new StreamPositionRange(StreamPosition.Start, uint.MaxValue),
+                cancellationToken)
+                .ConfigureAwait(false);
+            var sourceStreamEvents = lastUnProcessedEventPosition == 0
+                ? Enumerable.Empty<StreamEvent>()
+                : await _eventsFromStreams.FetchRange(
+                    filter.Scope,
+                    filter.Definition.SourceStream,
+                    new StreamPositionRange(StreamPosition.Start, lastUnProcessedEventPosition - StreamPosition.Start),
```

Comment:
Why lastUnProcessedEventPosition - StreamPosition.Start? StreamPosition.Start is alway == 0

Reply:
Could just be lastUnProcessedEventPosition